File: twosum/twosum.java
import java.util.Map;
import java.util.HashMap;
public class twosum {
    public static void main(String[] args){
        int[] nums = {1,2,3,4,5,6};
        int target = 9;
        int[] res = solution(nums, target);
        for(int i = 0; i < res.length; i++){
            System.out.print(res[i] + " ");
        }
    }
    private static int[] solution(int[] nums, int target){
            int[] res = new int[2];
            Map<Integer, Integer> map = new HashMap<>();
            for(int i = 0; i < nums.length; i++){
                if(map.containsKey(target-nums[i])){
                res[0] = i; res[1] = map.get(target-nums[i]);
                return res;
            }
            map.put(nums[i], i);
        } return res;
    }
}

File: twosum/twosum.py
def solution(nums, target):
    res = []
    map = {}
    
    for i in range(len(nums)):
        if (target-nums[i]) in map:
            res.append(map[target-nums[i]])
            res.append(i)
            return res
        map[nums[i]] = i
    return res

nums = [1,2,3,4,5,6,7,8,9,10]
target = 11
print(solution(nums, target))